ext/opentelemetry-exporter-cloud-trace/tests/test_cloud_trace_propagator.py

```diff
@@ -0,0 +1,123 @@
+)
+from opentelemetry.trace.span import (
+    SpanContext,
+    TraceFlags,
+    get_hexadecimal_trace_id,
+)
+
+
+def get_dict_value(dict_object: typing.Dict[str, str], key: str) -> str:
+    return dict_object.get(key, "")
+
+
+class TestCloudTraceFormatPropagator(unittest.TestCase):
+    def setUp(self):
+        self.propagator = CloudTraceFormatPropagator()
+        self.valid_trace_id = 281017822499060589596062859815111849546
+        self.valid_span_id = 17725314949316355921
+
+    def _extract(self, header_value):
+        """Test helper"""
+        header = {_TRACE_CONTEXT_HEADER_NAME: header_value}
+        new_context = self.propagator.extract(get_dict_value, header)
+        return trace.get_current_span(new_context).get_context()
+
+    def _inject(self, span=None):
+        """Test helper"""
+        ctx = get_current()
+        if span:
+            ctx = trace.set_span_in_context(span, ctx)
+        output = {}
+        self.propagator.inject(dict.__setitem__, output, context=ctx)
+        return output.get(_TRACE_CONTEXT_HEADER_NAME)
+
+    def test_no_context_header(self):
+        header = {}
+        new_context = self.propagator.extract(get_dict_value, header)
+        self.assertEqual(
+            trace.get_current_span(new_context).get_context(),
+            trace.INVALID_SPAN.get_context(),
+        )
+
+    def test_empty_context_header(self):
+        header = ""
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_valid_header(self):
+        header = "{}/{};o=".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), self.valid_span_id
+        )
+        new_span_context = self._extract([header])
+        self.assertEqual(new_span_context.trace_id, self.valid_trace_id)
+        self.assertEqual(new_span_context.span_id, self.valid_span_id)
+        self.assertEqual(new_span_context.trace_flags, TraceFlags(1))
+        self.assertTrue(new_span_context.is_remote)
+
+    def test_invalid_header(self):
+        header = "invalid_header"
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_invalid_span_id(self):
+        header = "{}/{};o={}".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), "0" * 16, 1
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_invalid_header_format(self):
+        header = "{}/{};o={}".format("123", "34", "4")
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_inject_with_no_context(self):
+        output = self._inject()
+        self.assertEqual(output, None)
+
+    def test_inject_with_invalid_context(self):
+        output = self._inject(trace.INVALID_SPAN)
+        self.assertEqual(output, None)
+
+    def test_inject_with_valid_context(self):
```

Comment:
A few more corner cases would be good here, e.g. 
* very large numbers to fill out the size of the trace id and span id
* zero span ID but valid trace ID and vice versa
* More flag values